pages/_app.tsx

```diff
@@ -29,6 +31,12 @@ const CustomApp = ({
   pageProps,
 }: ExtendedAppProps): JSX.Element | null => {
   const [user] = useState(pageProps.user);
+  const [featureFlags, setFeatureFlags] = useState(pageProps.featureFlags);
+
+  useEffect(() => {
```

Comment:
What does `useEffect()` allow us to do? e.g could we have run `setFeatureFlags(getFeatureFlags())` here and get the same result?

Reply:
It’s a hook (see docs link in a previous comment). Effectively it’s the React way of triggering behaviour based on some action. That could be rendering the component, a user interaction that triggers some state change, or any number of other things.